### PR TITLE
Stylized items, fixed some items and added item dependants (material for)

### PR DESCRIPTION
**Pull request by popara96**
- Loading classes on app start, default navigation set to character page instead of settings
- Fixed some item id's, added missing item (Dull Armor Piece) - some items still missing (Wind Glove, Pure Ignited Incinerator, Pure Light Synthesis)
- Added 'materialFor' field in item for purposes of displaying items in which the current item is used for crafting
- Updated all items' 'materialFor' fields
- Styled EvoItemRenderer to show item name and source (or sourceShort) in Items page for readability
- Added 'minimal' and 'border' options for EvoItemRenderer in order for them to stay minimal and borderless in stash preview
- Added navigate back button to ItemPage for easier navigation through items hierarchy

Screenshot 2024-04-14 150649
Screenshot 2024-04-14 150723

I'm new to TS so forgive me if some code is not following any convention, I tried to mimic the existing code style as much as I could. I'll be more than happy to rewrite some stuff based on feedback.

### CODE REVIEW

**Comment by paccao** [comment]
I have added the missing hyrule prophet items like Pure Light Synthesis and Pure Ignited Incinerator in https://github.com/ArgentumHeart/EvoHelper/pull/12

File: src/constants/items/forging/pieces.ts [hunk 1]
```diff
@@ -72,6 +89,7 @@ export const pieces: { [id: string]: EvoItem } = {
     effects: [],
     crafting: [],
     source: 'Blacksmith',
+    materialFor: ["Crystallized Emerald Blade", "Hand of God", "Tainted Neptune's Eye", "Hades' Artifact", "Tremor", "Paw of Lightning", "Darkforge Plate"]
```
Comment (paccao) [inline src/constants/items/forging/pieces.ts:92]:
```suggestion
    materialFor: ["Crystallized Emerald Blade", "Hand of God", "Tainted Neptune's Eye", "Hades' Artifact", "Tremor", "Darkforge Plate"]
```

not used for paw of lightning (according to ingame shop)

**Review by paccao — COMMENTED** [review]
Have reviewed 10 files where materialFor was added, will continue when I have time

File: src/constants/items/forging/ancientSoul.ts [hunk 2]
```diff
@@ -93,14 +96,15 @@ export const ancientSoulItems: { [id: string]: EvoItem } = {
...
-      "Hades's Artifact",
+      "Hades' Artifact",
       'Cursed Artifact',
       'Sealed Demise',
       "Harbinger's Essence",
       'Golden Fragment',
       'Diamond',
     ],
     source: 'Ancient Soul',
+    materialFor: ['Storm Sanctuary']
```
[the remark this answers is not in the document]
Comment (popara96) [reply src/constants/items/forging/ancientSoul.ts:107]: Magic Mirror is used for crafting Storm Sanctuary according to data in items, so Storm Sanctuary should be correct here.

**Comment by popara96** [comment]
> Have reviewed 10 files where materialFor was added, will continue when I have time

Thank you for your time looking through these, it does need proofreading since some item's id's are typed in wrongly, but I spoke to @ArgentumHeart and we came to the conclusion that this should (and will) be done dynamically in runtime instead of being hardcoded like this. So I will probably remove 'materialFor' and just leave the fixed typos here and there. Also pending is version 0.7.0. so we will get to this PR when it comes out.

**Comment by popara96** [comment]
I reverted most of what this originally was; now all that's left is
- fixed some typos
- added dull armor piece
- app starts on character page instead of settings
- refresh called on app start
- navigate back button in ItemPage

This will be pending until ***0.7.0.***

File: src/constants/items/forging/ancientSoul.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (paccao) [reply src/constants/items/forging/ancientSoul.ts]: Yep, my bad.

File: src/constants/items/forging/angelNature.ts
[the remark this answers is not in the document]
Comment (paccao) [reply src/constants/items/forging/angelNature.ts:1]: Woops, I mean Holy Bible ofc.

**Comment by popara96** [comment]
@ArgentumHeart are we reviving this or should we close the PR?

**Comment by ArgentumHeart** [comment]
I'm sorry I forgot about this until this tag. Looks fine to me, if conflict is resolved can merge asap @paccao

**Comment by popara96** [comment]
Resolved conflicts.

**Pull request merged**